### ISSUE TITLE
Regression after the skip registered task patch when registering the same var in two mutually exclusively ran tasks

### ISSUE THREAD

**Issue by srgvg** [ansible/ansible#2806]
Probably since #2685 

I have a play where I have 2 mutually exclusive tasks, which register the same variable; when the first task fires of, it's output gets overriden (.out gets unset) by the second tasks which is skipped.

```
4 - name: register root dir in nexus zip file
  5   shell: unzip -l ${tomcat_dir}/packages/${nexusfile.stdout} | awk \'{print $4}\' | cut -d/ -f1 | sort -u | grep -v -e^$ -e '^-' -e^Name | tail -1
  6   register: unzipdir
  7   only_if: 'not "solaris" in $group_names'
  8 
  9 - name: register root dir in nexus zip file (solaris)
 10   shell: bash -c "unzip -l ${tomcat_dir}/packages/${nexusfile.stdout} | grep / | cut -d/ -f1 | sort -u | /usr/xpg4/bin/grep -v -e^$ -e '^-' -e^Name -e^Archive | tail -1 " | awk '{print $4}'
 11   register: unzipdir
 12   only_if: '"solaris" in $group_names'
 13 
 14 - debug: msg=${unzipdir.stdout}
```

when running the not-solaris task:

```
TASK: [register root dir in nexus zip file (solaris)] ************************* 
skipping: [on-2-mgt]
skipping: [on-1-mgt]

TASK: [debug msg=${unzipdir.stdout}] ****************************************** 
ok: [on-2-mgt] => {"item": "", "msg": "${unzipdir.stdout}"}
ok: [on-1-mgt] => {"item": "", "msg": "${unzipdir.stdout}"}

TASK: [fail if we didn't detect the root dir in the zip file] ***************** 
skipping: [on-1-mgt]
skipping: [on-2-mgt]
```

when running the solaris task:

```
TASK: [register root dir in nexus zip file] *********************************** 
skipping: [int-on-arlon-mgt]
skipping: [int-on-aalst-mgt]

TASK: [register root dir in nexus zip file (solaris)] ************************* 
changed: [int-on-aalst-mgt] => {"changed": true, "cmd": "bash -c \"unzip -l /opt/tomcat/packages/intern-1.2.6.0.1-20130422.082144-18.zip | grep / | cut -d/ -f1 | sort -u | /usr/xpg4/bin/grep -v -e^$ -e '^-' -e^Name -e^Archive | tail -1 \" | awk '{print $4}' ", "delta": "0:00:00.029603", "end": "2013-04-29 15:33:31.149829", "item": "", "rc": 0, "start": "2013-04-29 15:33:31.120226", "stderr": "", "stdout": "intern-1.2.6.0.1-SNAPSHOT"}
changed: [int-on-arlon-mgt] => {"changed": true, "cmd": "bash -c \"unzip -l /opt/tomcat/packages/intern-1.2.6.0.1-20130422.082144-18.zip | grep / | cut -d/ -f1 | sort -u | /usr/xpg4/bin/grep -v -e^$ -e '^-' -e^Name -e^Archive | tail -1 \" | awk '{print $4}' ", "delta": "0:00:00.031622", "end": "2013-04-29 15:33:31.283959", "item": "", "rc": 0, "start": "2013-04-29 15:33:31.252337", "stderr": "", "stdout": "intern-1.2.6.0.1-SNAPSHOT"}

TASK: [debug msg=${unzipdir.stdout}] ****************************************** 
ok: [int-on-aalst-mgt] => {"item": "", "msg": "intern-1.2.6.0.1-SNAPSHOT"}
ok: [int-on-arlon-mgt] => {"item": "", "msg": "intern-1.2.6.0.1-SNAPSHOT"}
```

(PS: I'm testing my playbook with 1.2, but didn't migrate yet, which is why I'm still using pre 1.2 syntax)

**Comment by mpdehaan** [comment]
As discussed on IRC, I have no intention of fixing this one as I believe this to be the correction of a previous bug in the program.   Registration of variables is now reliable so you don't have to do the "is_set' kind of hijinks and I think it is a worthwhile change.

This:  " 12   only_if: '"solaris" in $group_names'"

should never be neccessary, and is a bit of an antipattern, and should be solved by a play like this, however tiny:

```
- hosts:  webservers:&solaris
  tasks:
      - include: ...

- hosts: webservers:!solaris
   tasks:
       - include: ....
```

# etc

Basically feel free to start lots of plays.  Use of group_by is also good to create dynamic groups so you don't have to know which boxes are Solaris boxes, if you want to do that.
